### PR TITLE
LPS-164919 Regenerate adaptive media images if content was lost

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

A significant amount of storage is consumed by adaptive media. This is a problem in particular for backups, as they require quite  some time to finish for large installations. Adaptive media content is a huge contributor to this.

Adaptive media images can be recreated from the original content with no data loss. By making the system more robust when stored data is not present, these images can be omitted from backups or removed if rarely used.

# How do I test this?

In a fresh portal, look at the contents of `data/document_library`:
```
$ find data/document_library -type d -name 'adaptive'
```
You should see the thumbnails and previews generated by adaptive media. Now, remove them:
```
$ find data/document_library -type d -name 'adaptive' -exec rm -rf {} \+
```
If you now go to documents and media widget in DXP, you'll see that document thumbnails are not displayed (you may need to upload some images first to see this). If you refresh the page after a couple of seconds, you'll see the thumbnails again. If you now look at the `data/document_library` directory again, the files should have been restored automatically.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
Some test failures are legit. I'll work on that and push-force the changes.

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
Pushed baseline commit again.

**Review by marcogalluzzi — APPROVED** [review]
Had a look to the code, run automated tests, executed manual test instructions. All is fine.

**Review by AliciaGarciaGarcia — APPROVED** [review]

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/validator/AMImageValidatorImpl.java [hunk 1]
```diff
@@ -95,6 +122,12 @@ public boolean isValid(FileVersion fileVersion) {
 		return true;
 	}
 
+	protected void setAMImageEntryLocalService(
```
Comment (AliciaGarciaGarcia) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/validator/AMImageValidatorImpl.java:125]: It's more of my taste to use ReflectionTestUtil than create a method only for testing, but LGTM

**Pull request closed without merging**